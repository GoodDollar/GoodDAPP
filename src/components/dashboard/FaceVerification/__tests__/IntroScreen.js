// eslint-disable-next-line import/order
import { initUserStorage } from '../../../../lib/userStorage/__tests__/__util__'
import React from 'react'
import renderer from 'react-test-renderer'
import { withThemeAndLocalizationProvider } from '../../../../__tests__/__util__'
import ImportedIntroScreen from '../screens/IntroScreen'

const IntroScreen = withThemeAndLocalizationProvider(ImportedIntroScreen)
const screenState = {
  isValid: true,
}

jest.setTimeout(30000)

describe('FaceVerification IntroScreen', () => {
  beforeAll(async () => {
    await initUserStorage()
  })

  it('renders without errors', () => {
    const tree = renderer.create(<IntroScreen screenProps={{ pop: () => {}, screenState }} />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = renderer.create(<IntroScreen screenProps={{ pop: () => {}, screenState }} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
