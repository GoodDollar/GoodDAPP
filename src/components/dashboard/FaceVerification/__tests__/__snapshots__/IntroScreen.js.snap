// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaceVerification IntroScreen matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
>
  <div
    className="css-view-1dbjc4n r-display-6koalj r-flexDirection-eqz5dr r-flexGrow-16y2uox r-maxHeight-87ditf r-overflowX-lltvgl r-overflowY-1rnoaur r-padding-edyy15 r-position-bnwqim r-width-13qz1uu"
    data-name="viewWrapper"
    style={
      Object {
        "backgroundImage": "linear-gradient(3.141592653589793rad,#00AFFF,#2DC0F7,#28C0EF,#23C0E7,#1EC1DF,#19C1D7,#14C1CF,#0FC2C7,#0FC2C7,#0AC2BF,#05C2B7,#00C3AF)",
      }
    }
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-14lw9ot r-borderRadius-1f0042m r-display-6koalj r-flexDirection-eqz5dr r-flexGrow-16y2uox r-flexShrink-1q142lx r-justifyContent-1777fci r-paddingBottom-1l7z4oj r-paddingHorizontal-1e081e0 r-paddingLeft-1m04atk r-paddingRight-1pyaxff r-paddingTop-95jzfe r-paddingVertical-1yzf0co"
    >
      <div
        className="css-view-1dbjc4n r-flexGrow-16y2uox r-justifyContent-1wtj0ep r-width-13qz1uu"
      >
        <div
          className="css-text-901oao r-color-cddv6f r-fontFamily-vnw8o6 r-fontSize-1x35g6 r-fontWeight-b88u0q r-lineHeight-17rnw9f r-marginTop-1wzrnnt r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
          dir="auto"
        >
          ,
          <span
            className="css-text-901oao css-textHasAncestor-16my406 r-color-cddv6f r-fontFamily-vnw8o6 r-fontSize-1x35g6 r-fontWeight-1it3c9n r-lineHeight-17rnw9f r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
            dir="auto"
          >
            
Verify you are a real
live person
          </span>
        </div>
        <div
          className="css-text-901oao r-color-cddv6f r-fontFamily-vnw8o6 r-fontSize-1i10wst r-fontWeight-1it3c9n r-letterSpacing-1j4bhhx r-lineHeight-waer2z r-marginTop-1wzrnnt r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
          dir="auto"
          letterSpacing={0.18}
        >
          Your image is only used to prevent the creation of duplicate accounts and will never be transferred to any third party
        </div>
        <div
          className="css-text-901oao r-color-1q5uqvh r-cursor-1loqt21 r-fontFamily-vnw8o6 r-fontSize-1i10wst r-fontWeight-b88u0q r-lineHeight-eaezby r-marginTop-1x0uki6 r-textAlign-q4m81j r-textDecorationLine-1ddef8g r-textTransform-3twk1y"
          data-focusable={true}
          dir="auto"
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex="0"
        >
          Learn More
        </div>
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-marginBottom-10qg9sp r-marginTop-1x0uki6 r-width-13qz1uu"
        >
          <FashionPhotoshoot.svg />
        </div>
        <div
          className="css-view-1dbjc4n r-display-6koalj r-justifyContent-1777fci r-width-13qz1uu"
          style={
            Object {
              "WebkitAlignItems": "stretch",
              "WebkitBoxAlign": "stretch",
              "alignItems": "stretch",
              "backgroundColor": "rgba(0,0,0,0.12)",
              "borderBottomColor": "rgba(0,175,255,1.00)",
              "borderBottomLeftRadius": "50px",
              "borderBottomRightRadius": "50px",
              "borderBottomStyle": "solid",
              "borderBottomWidth": "0px",
              "borderLeftColor": "rgba(0,175,255,1.00)",
              "borderLeftStyle": "solid",
              "borderLeftWidth": "0px",
              "borderRightColor": "rgba(0,175,255,1.00)",
              "borderRightStyle": "solid",
              "borderRightWidth": "0px",
              "borderTopColor": "rgba(0,175,255,1.00)",
              "borderTopLeftRadius": "50px",
              "borderTopRightRadius": "50px",
              "borderTopStyle": "solid",
              "borderTopWidth": "0px",
              "marginBottom": "0px",
              "marginLeft": "0px",
              "marginRight": "0px",
              "marginTop": "0px",
              "minHeight": "44px",
              "minWidth": "64px",
              "msFlexAlign": "stretch",
              "paddingBottom": "0px",
              "paddingLeft": "0px",
              "paddingRight": "0px",
              "paddingTop": "0px",
            }
          }
        >
          <div
            aria-disabled={true}
            className="css-view-1dbjc4n r-flex-13awgt0 r-overflow-1udh08x r-position-bnwqim"
            disabled={true}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
            style={
              Object {
                "WebkitAlignSelf": "stretch",
                "alignSelf": "stretch",
                "borderBottomLeftRadius": "50px",
                "borderBottomRightRadius": "50px",
                "borderTopLeftRadius": "50px",
                "borderTopRightRadius": "50px",
                "msFlexItemAlign": "stretch",
              }
            }
          >
            <div
              className="css-view-1dbjc4n r-alignItems-1awozwy r-flex-13awgt0 r-flexDirection-18u37iz r-justifyContent-1777fci"
              style={
                Object {
                  "WebkitAlignSelf": "stretch",
                  "alignSelf": "stretch",
                  "msFlexItemAlign": "stretch",
                }
              }
            >
              <div
                className="css-view-1dbjc4n r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr"
              >
                <div
                  className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-flex-13awgt0 r-flexDirection-18u37iz"
                >
                  <div
                    className="css-text-901oao r-alignItems-1awozwy r-color-jwli3a r-fontFamily-vnw8o6 r-fontSize-ubezar r-fontWeight-majxgm r-justifyContent-1777fci r-letterSpacing-oxtfae r-lineHeight-hbpseb r-paddingTop-njp1lv r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-tsynxw"
                    dir="auto"
                  >
                    OK, VERIFY ME
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
